include/extractor/internal_extractor_edge.hpp

```diff
@@ -142,14 +142,16 @@ struct InternalExtractorEdge
     }
 };
 
-struct CmpEdgeByInternalStartThenInternalTargetID
+struct CmpEdgeByInternalStartThenInternalTargetThenNameID
 {
     using value_type = InternalExtractorEdge;
     bool operator()(const InternalExtractorEdge &lhs, const InternalExtractorEdge &rhs) const
     {
         return (lhs.result.source < rhs.result.source) ||
                ((lhs.result.source == rhs.result.source) &&
-                (lhs.result.target < rhs.result.target));
+                (lhs.result.target < rhs.result.target ||
+                 (lhs.result.target == rhs.result.target &&
+                  lhs.result.name_id > rhs.result.name_id)));
```

Comment:
Well... fixes. It randomly chooses any of the non-empty names over an empty name. In that case, your test-case is still wrong, since it offers multiple possible names (50:50 chance)